tools/cli/factory.go

```diff
@@ -45,6 +45,15 @@ type WorkflowClientBuilderInterface interface {
 	BuildAdminServiceClient(c *cli.Context) (adminserviceclient.Interface, error)
 }
 
+// AuthorizationManagerInterface is an interface to build permission checking layer to cadence service.
+// User can customize builder by implementing this interface, and call SetAuthorizationManager after initialize cli.
+// (See cadence/cmd/tools/cli/main.go for example)
+// The customized AuthorizationManager may have more processing on Env, Address and other info.
+type AuthorizationManagerInterface interface {
+	//FilterUnauthorizedOperations decides which operation will be available in this CLI
+	FilterUnauthorizedOperations(*cli.App) *cli.App
```

Comment:
How does it do it? From the comment it is not clear at all. What about when an operation is allowed, but only for objects (like domain) that user owns?